### PR TITLE
improvement/highlight-over-plan-accounts

### LINKED ISSUE #256: Highlight Sales dashboard rows for customers outside their seat limit

### PR DESCRIPTION
**Pull request by dabeeeenster**

### CODE REVIEW

File: api/organisations/models.py [hunk 1]
```diff
@@ -70,6 +70,15 @@ def over_plan_seats_limit(self):
...
         )
 
+    def over_plan_and_free_seats_limit(self):
+        if self.has_subscription():
+            return (
+                self.has_subscription()
+                and 0 < self.subscription.max_seats < self.num_seats
+            )
+        else:
+            return self.num_seats > 1
```
Comment (dabeeeenster) [inline api/organisations/models.py]: Should maybe make this a constant @matthewelwell ? not sure if that value is defined anywhere?

Comment (matthewelwell) [reply api/organisations/models.py]: Yep, we can probably just put it on the `Subscription` class or something.

File: api/organisations/models.py [hunk 2]
```diff
@@ -70,6 +70,15 @@ def over_plan_seats_limit(self):
             self.has_subscription() and 0 < self.subscription.max_seats < self.num_seats
         )
 
+    def over_plan_and_free_seats_limit(self):
```
Comment (matthewelwell) [inline api/organisations/models.py]: Can't we just refactor the method above to:

```python
def over_plan_seats_limit(self):
    return self.num_seats > getattr(self.subscription, "max_seats", MAX_FREE_SEATS)
```

This mew method feels like serious duplication.

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: api/organisations/models.py (hunk 1 above)
Comment (dabeeeenster) [reply api/organisations/models.py]: Added

File: api/organisations/models.py (hunk 2 above)
Comment (dabeeeenster) [reply api/organisations/models.py]: Tidied up

File: api/app_analytics/influxdb_wrapper.py [hunk 3]
```diff
@@ -214,6 +217,14 @@ def get_top_organisations(date_range: str, limit: str = ""):
     dataset = {}
     for result in results:
         for record in result.records:
-            org_id = int(record.values["organisation"].partition("-")[0])
-            dataset[org_id] = record.get_value()
+            print(record.values["organisation"].partition("-")[0])
```
Comment (matthewelwell) [inline api/app_analytics/influxdb_wrapper.py]: Why is this here?

File: api/organisations/models.py (hunk 2 above)
Comment (matthewelwell) [reply api/organisations/models.py]: What's the reason for not going with my suggestion? I don't really see the need for the new method at all...

Comment (dabeeeenster) [reply api/organisations/models.py]: getattr doesnt work for our needs if the subscription object is missing but I've removed the additional method

**Review by matthewelwell — APPROVED** [review]
